In [1]:
import pandas as pd
import numpy as np
from random import sample
from scipy import stats
import math

### Part 1
Use the random sampling method to generate a random sample of size 45 from the given International Migration Rates. 

**Then calculate the population mean, the variance and standard deviation of the population.**

In [2]:
population = [-0.2,0.9,2,2.1,0.3,2.7,0.4,1.8,0.5,0.1,4.1,0.9,0.1,0.7,1,0.3,1.8,0.6,0.1,5.2,1.2,7.8,-0.2,4.5,4.5,4.2,0.4,0.2,0.6,15.2,1.9,0.6,1.2,1.4,0.4,0.1,3.2,2,0.6,0,0.2,0.1,0,0.7,0,0.3,-0.1,2.7,0.3,0.8,0.8,1.1,-0.2,2.7,1.5,0.7,1.9,1.9,0.5,0.3,1.1,0.1,0.2,-0.1,0.5,4.3,0.6,-0.1,0.7,9.8,0,0.6,0.8,0.8,1.7,0.4,-0.2,0.8,-0.9,0.9,0.4,-0.1,4.9,6,0.6,0.2,0.1,1,0.3,1.2,0.3,0.9,1,1.2,0.3,0.1,0.8,0.6,3.7,4.3,0.3,0.2,0.1,2.8,-0.1,-0.3,0.8,1.4,0,0.8,0.9,0.8,0.1,1.6,4.8,0.8,0.4,0.1,0.2,0.1,-0.2,0.4,1.3,0.2,0.3,9,0.1,0.8,0.4,0.2,1.4,0.8,0.2,1,0.3,0.3,0.8,0.1,2.7,-0.3,1.4,-0.2,0.4,3.1,0.2,0.3,-0.1,5.6,0.2,1.6,-0.2,6.5,1.2,0,4.2,2.3,1.6,1,2.8,1.4,0.2,3,0.9,1.5,0.1,0.4,-0.2,0.5,2.4,0,1.1,1.8,0.2,2.5,1.3,-0.1,1.1,0.4,2.2,0.5,0,0.6,-0.1,0.4,-0.2,-0.6,-0.1,0.9,2.1,4.1,-0.2,0.4,1.9,0,2.7,0.3,1.9,1,0.7,0.7,-0.1,0.1,0.6,0,0.4,0.4,1.5,1.2,-1.2,-0.1,2,7,0,-0.1,1.9,2.5,-0.1,2.8,1.8,13.3,-0.3,3.6,0.6,3,1.7,9.5,-0.1,0.5,-0.1,2,1.2,1.6,0,1,0.2,0.2,0.5,3.6,0.3,0.3,1.6,4.1,3.3,0.9,0.1,1.9,1.7,0.8,0.4,0.2,0.9,0.8,1.6,2.1,0.1,0.2,0.4,0.8,0.4,0.2,0.1,2.2,0.7,1.4,-0.3,0,1.1,0.7,2,1,0.7,0.6,0.8,-0.1,0.8,0.3,0.1,0.7,0.3,2.4,0.2,0.5,1.9,0.5,0.6,5.6,0.6,0.6,0.3,0,0.9,1.5,3.8,2.5,3.6,1.6,-0.3,0.9,0.1,0.6,2.3,0.8,-0.1,1.4,1.5,0.2,0.2,0.2,-0.1,0.2,2.3,0.4,2.1,1.7,0.3,0.5,0.3,0,0.4,-0.1,0.4,0.1,0.2,0.4,0.3,0.4,0.2,0.1,0.6,0.7,0,-0.3,0,4.3,3.2,-0.2,0.3,0.6,4.5,-0.2,0.6,1.2,0.8,1.9,0.4,0.8,1,1,0.7,-0.1,7.4,1,-0.1,0.5,2.9,9.4,0.1,1.1,0.7,1,2.3,0.1,9.5,5.3,2.8,0.5,7.2,0.3,1.8,0.5,0,0,1.1,0.9,0.4,0.3,0.8,0.5,-0.3,0.1,0.6,0,0.8,1.6,0.4,3,-0.1,0.4,0.7,2.1,0.1,1.2,1.8,0.5,1.2,2.2,0.3,0.3,4.2,-0.2,0.2,0.4,0,0.2,0.1,1.7,0.8,0.6,0.7,0,-0.1,0.6,0.1,0.1,0.9,0.7,-0.1,0.7,0,3.3,0.1,0.8,0.6,0.2,3.9,0.1,0.4,0,0,-0.1,1.6,0.7,0,-0.1,0.4,1.5,0.9,0,1,1.5,0.7,3.4,0.3,0.8,-0.3,0,0.5,0.8,2.4,0.9,0,0.2,1.4,1.1,-0.3,1.7,0.4,0.7,0,4.3,1.8,-0.1,1.5,0.4,1,1,-0.3,0.1,1.3,1.1,11,0.7,-0.1,7.2,5.4,2.9,2.8,1.8,1.7,0.6,1,1.8,0.3,0.5,0.5,-0.1,0.4,0.7,0.9,1.5,1,0,0.2,4.2,1,0,0.2,3.9,1.2,0.6,0.1,0.6,0.3,1.1,0.5,2.1,0.3,1,-0.1,0.5,1.3,1.7,0.6,0.4,1.6,1.5,1.3,1.4,2.3,0.2,-0.1,3.2,0.6,-0.2,0.2,0.4,0.5,0.8,0,0.7,0,0.4,0.1,0,0.7,0.4,1.4,1,2.9,2.3,3.5,2.8,0.6,0.1,1.6,-1.1,-0.1,0,2.4,3.2,-0.4,1.8,2.4,0.1,0,3.4,-0.4,5.6,3.5,1.1,0.3,0.1,0.2,4.3,0.4,5.8,0.5,0.2,1.2,0.6,-0.1,0.5,5.8,2.7,0.1,0.1,2,-0.3,0.1,1.6,1.3,1,1.3,7.6,2.1,-0.1,-0.6,3.5,-0.1,0.5,1.7,0.6,1.7,0.4,3.6,0.1,0,0.6,3.3,3.2,0.9,0.1,1.9,3.3,0.3,0,0.1,0.2,0.3,-0.1,0,2.3,3.2,1,6,0,0.3,0.8,1.1,0.4,1.4,0.4,0,1.4,0.9,1.2,1.6,1.3,0.4,5.8,0.3,-0.3,3.4,1.1,0.1,1.6,0.8,0.4,1.6,0.7,0.6,0.1,0.8,1.1,3.3,2.9,5.5,2.9,3.7,0.4,0.6,1.4,0,2.8,-0.1,1.3,0.9,1.1,2.1,1,-0.1,0.1,0.2,0.6,0.3,0.7,1.1,1.2,3.7,0.1,0.4,0.1,1,0.4,1.8,0.9,0.4,3.6,-0.2,0,0.4,0.8,4.1,0.3,3.4,0,1.4,6.8,8.3,1.8,0.5,0,0.1,2.4,-0.1,1.3,11,9.4,6.5,1.3,0.8,5.8,1.2,-0.4,0.6,0,0.9,0.6,0,2,1.3,0,0.3,4.9,0.6,0.6,0.6,1.2,0.3,0,0.1,1.2,5.9,2.1,-0.3,0,0.6,1.6,-0.2,0.4,0.3,-0.1,0,1.8,-0.1,9.7,1.4,1.8,1.6,6.9,0.1,1.7,0.4,0.6,1.3,1.4,-0.4,0.5,0.3,0.6,1,2.7,0.5,0.2,0.9,0.7,5.1,1.5,1.3,0,-0.1,1,2.4,-0.1,3.2,1,3,0.6,0.7,-0.1,0.3,0,2.1,10.7,-0.2,4.3,0.4,-0.1,0.6,0.5,0.4,1.8,-0.1,6.9,0.7,0.3,2.1,0.7,1.8,0,0,0.1,1.6,0,1,0.8,0.5,0.5,0.6,0.4,0.8,-0.2,3,0,0.1,0,0.4,0.3,1.2,0.5,2.8,2.6,0.3,2.5,0,0.2,1.4,0.2,2.2,0.5,0.9,-0.1,0.4,-0.1,1,1.8,1.4,1,2.2,1.2,0,-0.1,2.9,0.3,0.1,0.2,-0.1,0.1,2,0.7,-0.2,0.4,-0.1,0.4,0.9,0.8,0.6,0.2,1.3,0,0.5,1.2,0.1,2.7,0.3,2.4,0,0.2,1.2,0,0.8,1.6,2.9,0.1,0.3,2.8,0.5,-0.2,2.3,1.7,2,-0.1,1.2,0.1,0,0,0.9,0.2,3.3,0.6,-0.2,0.5,4.2,0.7,-0.1,1.4,0.6,-0.1,1.2,5.9,2.1,0.8,1.9,2.5,2.8,1.4,6.6,5.2,-0.3,1.9,0.3,0.3,0.3,-0.1,1.3,0,1.7,0.5,0.8,-0.2,2.7,0,1.1,1.2,0.6,0.7,0.1,2.7,1.1,0.1,1.3,0,7.5,0,0.5,1.1,1.3,0,1.8,-0.2,9.8,0.7,0.1,5.6,1.5,0.7,3.3,-0.1,1.4,-0.2,0.3,-0.1,-0.3,0,0.3,-0.2,0.8,1.4,7.1,5.4,5.4,0.2,1.1,0.6,0.2,0.1,0.6,-0.1,0.2,0.4,0.2,0.6,1.2,1.6,0,3.5,0.3,3.2,0.3,1.1,6.6,2.9,0,1.2,0.1,0.9,1.4,0.1,-0.1,-0.1,1.2,4.9,1.8,-0.4,2.1,0.6,-0.2,1.1,0.2,4.7,0.3,2,0.3,0.7,1.5,0.3,5.4,0.4,2.3,0.3,0.4,16.5,0.2,-0.4,0.4,1.1,0.1,1.2,1.2,0.6,0.5,0.5,0.6,1.1,1.3,0.3,0,3.1,0.6,2.6,1.2,6.7,0.9,0.1,0.4,-0.4,0.3,1.3,0.1,0.2,-0.3,0.6,3.4,0.6,0.7,0.9,0,0.5,0.9,0.5,0.1,1.1,0,0.1,4.2,2,0,0,0.5,1.4,2.1,1.6,0.7,1.9,-0.2,-0.1,0.8,0.2,0.7,0,1.5,1,0,0.3,1.1,0,0.8,-0.2,0.1,0.1,1.1,0.3,0.8,-0.1,0.5,18.6,0.1,1.3,0.4,0.1,3.4,0.3,0.5,0.8,-0.2,0.8,0.2,0.7,3.1,0.4,0.6,0.8,0,0,4.5,-0.2,3.1,0.8,2.6,1.2,0.6,2.2,7.2,0,1.3,2.3,0,0.6,-0.2,0.4,0.8,0.4,0,0.9,0.1,0.8,-0.1,0,2,0.2,1.2,2.2,0.3,0.6,0.6,5.3,1.8,0.8,1.1,5.4,0.9,0.4,1.1,1.2,1.6,0.6,0.4,1.6,-0.1,2.7,0.5,0.9,0.5,2.3,0.2,0.2,2.6,1,8,-0.1,0.5,0.6,0.9,9.5,0.6,1.9,0.3,3.3,0.8,0.1,1.8,0.2,-0.2,1,0.1,1.5,3.8,1,-0.2,0.7,0.8,0.4,2.8,-0.2,0,3.7,-0.1,0.1,1,0.2,2.2,1.5,0.2,0,2.9,2.5,0.6,1.1,1.2,1.1,1.4,0.2,0.2,1.9,-0.2,0.2,1.2,0.5,2.4,4,3.1,1.1,1.6,1.8,0.8,0.2,0.6,0,0,1.4,-0.1,1.2,0.6,1.4,0.9,3.7,0.4,0.2,-0.1,0.8,0.1,0.4,0.4,0.6,-0.1,0.3,0.1,0.4,0.9,0,0,-0.3,0.3,-0.2,0.4,1.8,0.2,2,0,0.3,4.9,3.4,4.3,1,0.2,0.2,0.2,1.4,0,0,-0.4,0.8,0.2,6.4,0.4,1.2,2.9,0.2,-0.1,0.1,0.4,-0.1,-0.2,0.9,-0.1,0.5,0.7,1.6,1.2,0.9,0.5,1,2.4,-0.2,0,0.4,0.3,0.3,-0.3,-0.2,0.4,-0.3,0.2,1.1,0.2,-0.1,0.2,0.1,-0.2,-0.6,4.4,1.7,0.5,0,1.8,-0.3,1.4,0.4,0.5,0.3,1.7,0.4,0.1,1.9,1.1,0.2,-0.2,0.6,0,-0.1,1.6,1,2.1,4.5,4.8,0.3,6.5,3.9,0.2,0.8,-0.1,1.4,1.4,1.6,1.2,0.3,0.2,0.7,3,0.6,4.5,0.1,3.5,3.8,-0.3,1.7,3.2,0.4,0.9,0.7,0.1,1.1,2.5,0.1,1.7,1.4,0.9,0.9,1.1,0.8,0.2,0.4,0.4,1.2,-0.1,1.6,0.9,2.4,0.5,0.9,0.6,1.1,0.6,0,0.2,1.5,1.4,-0.1,6.8,9.8,3.4,0.5,0.7,0.7,0.4,2.6,3.3,0,0,0.4,2,0.5,0.4,3.4,0.7,0.6,0.4,-0.1,0.2,1,1,0.2,-0.5,0.2,1.5,0.4,0.6,0.1,0.8,4.2,0,1,0.1,0.1,1.3,0.6,0,1.4,0,-0.3,-0.1,0,0.4,0,-0.1,0.4,1.8,-0.3,0.1,0.4,0.7,4.9,0.7,5.4,0.1,0.3,0.2,0,0.7,2.9,2.7,1.3,1,-0.4,1.1,0,0.3,0.4,0.6,1.4,3.7,1.7,0.4,1.2,-0.1,0.5,0.5,0.7,4.8,0.5,1.6,0.5,0.1,1.4,0,-0.2,-0.2,0.9,0,1.5,0.3,0.9,8.6,0.3,0.4,0,0.1,2.3,-0.3,0.1,0.6,0.3,0.5,19.1,2.1,8.3,9.9,0,2.2,2.6,0.5,1,1.5,0.3,-0.5,-0.1,-0.1,3.1,0.3,0,1.4,3.9,0.8,0.3,-0.4,0.8,0.3,0.4,1.3,1.2,0.5,-0.5,1.1,2.1,3.2,3.9,-0.1,6.6,-0.3,-0.1,6.8,0.9,0.5,0,4.1,-0.1,2,1.4,0.1,0.8,0.7,0.1,2.4,-0.8,1.6,0.9,0,1.5,1.8,-0.2,10.5,0.4,0.4,0.9,2.9,2.1,3.5,1.3,2.5,6.9,2.4,-0.1,0.8,8.8,0.2,0.8,3.3,0,0.9,0.3,0.1,-0.2,0.1,0,0.7,-0.1,4.5,1.1,0.1,0.1,0.4,0.2,0.1,0.7,2.6,0,4.3,0.9,0.3,2.3,2,0.3,0.7,0.3,1.9,7.5,2.3,0.9,-0.2,3.9,0.5,0.4,0.5,0.4,-0.1,1.2,0,0.7,-0.2,0.3,3.8,0.9,5,0.7,2.7,0.2,11.2,0.4,1.2,2.6,2.7,0.7,1.3,0.6,0.1,0.5,0.2,1,-0.1,1.1,0,0.1,-0.1,10.3,0.8,0.5,2.7,5.3,1.1,2.5,1.7,0.3,0.8,1,0.1,1.5,2,-0.2,4.6,-0.2,0.4,1.1,1.6,2.9,0.3,1.3,0.5,0.5,-0.1,0.7,0.1,0.6,0,1,2,1.6,0.8,3.5,0.1,2.2,0.3,4.5,3.9,0.4,4.5,-0.3,1.4,4.5,10.3,0.4,5.5,2.1,-0.1,1.4,0.6,0.5,-0.3,0.9,0.2,2.5,0.1,0.2,0.1,15.8,0.6,0.7,0.1,5.7,3,0.7,1.1,1.8,0,0.4,1.6,0.4,0.7,1.2,0.1,0,0,0.8,0.2,0.8,0.6,1,0,7.5,0.8,0.6,0.5,1.2,0.1,0.2,0.7,0.1,0.5,2.1,1.6,8.2,0.1,0.6,0.4,0.5,-0.9,5.4,2.4,-0.4,0.1,-0.2,-0.4,0.6,0.2,1.8,0.2,0.6,2.2,-0.4,0,0.6,-0.1,0,0.1,-0.2,0.2,0.3,-0.1,2.5,1.3,5.2,6.5,0.1,0.2,0.2,0.3,0.4,2.1,0,1.6,1.2,3.3,0.6,0.1,0,0.2,0.4,-0.1,0.3,0,1,2.5,2.7,0.1,3.3,0.1,4.5,1.3,0,0.1,1.3,1,2.2,1.7,0.4,1.7,2.9,0.6,0.4,0.6,4.7,1.6,3.1,1.5,0.2,0.2,1.3,-0.4,2.9,0.4,-0.3,1.5,1.5,0.1,-0.1,3.4,0.9,0.7,1,0.2,0.5,0.4,0.8,0.2,5.8,0.2,1.7,0.4,0.4,0.1,0.5,0.2,1.6,1,1.4,7.2,5.2,6.8,1.5,0.3,2.1,-0.2,0.3,0.7,3.3,0,0.7,0.4,0.6,1.2,2.1,0.5,4.8,0.2,1.5,12.4,0.9,0.6,2.1,1.1,0.7,0,7.1,0,1.4,-0.4,-0.7,0,0,0.1,0.6,0.4,0.8,1.2,-0.3,-0.1,0.1,0.8,0.4,7.8,-0.1,2.3,0.8,1.2,2.5,0.6,-1.5,0.3,0.2,1.9,0.6,-0.1,2.9,1.3,0.2,3.1,7,0.8,-0.7,1.1,0,1.6,0.1,0.1,5.2,2,0.4,0.9,1.2,0.7,0.7,3.8,1.2,0.1,1.9,0.4,0.7,1.6,4.7,0.9,0.3,0,-0.2,1.8,0.5,0.6,0.4,2.7,0.2,0.5,0.5,0.1,0.1,1.6,-0.1,1,0.4,0.3,2.2,0.4,-0.4,0.7,3.7,-0.2,1,1.9,0.7,0.5,0.1,0.5,1,7,2.9,4,2.9,0.2,1.4,4.6,9.4,-0.5,2.6,0.1,0.7,1.4,1.1,7.3,2.6,0.9,-0.2,-0.1,0.1,1.2,0.2,1,3.7,10.7,2.3,1.9,1,0.1,2.5,1.7,1.2,0.8,0.6,0.1,3.1,0.3,0.6,0.6,0.5,0.5,0,1.5,0.8,0.6,2,0,1.7,-0.2,0.3,1.7,0.1,0.8,1.2,2.1,4.5,0.2,0.6,0.7,-0.1,0.8,0.6,2.5,0.2,5.4,0.5,0.2,1.5,0.3,1.2,1.2,1,0.9,0.2,0.2,3,1.8,1.7,2.7,1,0.6,1.1,0.2,0.5,1.3,-0.1,2,0.3,0.2,1.8,0.5,1.6,0.4,0,0,1.8,0.6,3.4,0.9,2,2.6,1.5,0.5,6.2,0.6,1.9,0.3,0.2,0.7,0,-0.1,1.5,1.4,1.9,1.5,1.1,0.3,0.2,0.7,0.5,0.7,0,0.4,0,2,1.1,2.5,0.9,-0.1,1.7,3.1,1,2.2,3.4,1,2.3,1.3,1.2,2,2.1,0.8,1.6,0.5,-0.1,3.5,1.4,1.2,3.8,0.4,0.2,0.6,0.7,0.6,0.6,0.7,-0.2,-0.1,0.1,-0.7,0.4,0.1,9.4,1.6,0.3,1,13.8,3.3,0.1,0.3,3.3,0,-0.5,2,2,2,0.6,0.5,2.8,0.3,0.3,1.1,0.3,0.5,0.1,1.4,0.7,0.4,3.3,0.4,0.1,0.3,1.9,1.3,0.4,-0.2,0,0.2,1.1,0.9,1,3.7,0.1,0.4,0.9,2.4,1.9,0.2,0.2,0.3,0,0,3.4,-0.5,1.3,2,1.7,0.4,0,4.3,0.6,2.1,0.6,1,-0.3,0.1,-0.1,9.6,0.2,0.6,0.5,2.9,0.2,0.1,-0.2,3,0.2,11,0.5,0.5,3.4,0.2,0.5,5.4,1.4,0.3,0.2,-0.3,0.9,0.2,0,1,2.8,0.7,1.4,1.7,0.6,0.1,2.7,0.4,0,0.5,2.5,0.1,0.1,-0.2,1,0.2,1.3,0.1,-0.2,1.7,0.7,8,1.4,0.6,1.2,0,-0.4,0.3,1.1,0.6,-0.2,-0.2,1.2,0.7,2.5,0.6,-0.8,0,0.9,0,1.2,0.2,0.9,0,2.4,0.4,0.8,0.6,0.2,0.1,1.3,0.8,1.3,0.7,0,2.4,2.8,2.3,0.3,0.2,0.2,0.4,3.9,0.3,1.5,0.4,-0.3,2.2,0.6,1.4,2.6,-0.3,0.7,0.8,0.5,3.8,2.6,1.7,1,0.2,2.4,5.4,0,0.6,-0.3,2.7,0.8,2.2,0.2,1.2,0.1,1.2,2.4,1,0.5,0.5,1.1,2.6,0.2,0,0.3,0.2,0.1,-0.1,2.4,1.9,0.8,1.7,0,0.6,0.5,0.8,0.4,0.2,1.7,3.6,0.6,0.7,1,1.9,-0.2,0.1,0.2,0.7,-0.3,0.3,1.3,6.2,0.5,2.7,1.4,0.4,0.7,0.8,0.8,0.4,0.5,2.5,-0.1,-0.2,2.3,0.7,0.9,0,0.1,0,1.9,0.8,4.8,1,1.3,1.1,-0.1,0.4,0.7,0.7,0.5,0.3,0.1,5.4,0.2,-0.2,0,0.3,0.6,1.9,0.7,0.8,0.5,1.1,0.1,0.2,1.3,0.4,0.2,8.1,2.5,4.6,1.3,0,0.7,1.1,1.5,0.2,-0.2,9.3,0.5,1,2,2.3,1.3,0.7,0.5,5.5,2.6,0.9,0.1,-0.1,0,1.6,0.7,0.1,1.5,0,-0.1,0.8,0.2,1.6,0,0.2,-0.2,2.2,0.4,0.3,2,5.1,0.9,0,1.5,0.6,-0.1,0.2,4.5,0.4,0.4,0,0.1,0.8,0.5,0,0.9,0.1,0.2,0.7,0.3,0.7,1.5,0,6.3,0.6,0.7,1.6,0.6,2.3,0.8,1.5,3.8,5.2,1.8,3.3]

In [3]:
population = np.array(population)

In [4]:
population_45 = np.random.choice(population, size = 45, replace = True)

In [5]:
print("The sample mean is %.5f"%np.mean(population))

The sample mean is 1.24385


In [6]:
print("the sample variance is %.5f"%np.var(population))

the sample variance is 3.60042


In [7]:
print("the sample standard deviation is %.5f"%np.std(population))

the sample standard deviation is 1.89748


### Part 2
Using the random sample from Part 1, complete the following analysis of the data.

**1. Calculate the sample mean, sample standard deviation, sample size and sampling (standard) error**

In [8]:
print("The sample mean is %.5f"%np.mean(population_45))

The sample mean is 1.12000


In [9]:
print("the sample variance is %.5f"%np.var(population_45))

the sample variance is 2.18693


In [10]:
print("the sample standard deviation is %.5f"%np.std(population_45))

the sample standard deviation is 1.47883


In [11]:
print("the sample size is %.5f"%population_45.size)

the sample size is 45.00000


In [12]:
z_score = np.zeros(len(population_45))
z_score = (population_45 - np.mean(population_45))/np.std(population_45)

In [25]:
sampling_error = (np.std(population))/(math.sqrt(population_45.size))
print("the sampling error is %.5f"%sampling_error)

the sampling error is 0.28286


**2. Use the sample to create 92%, 96% and 98% confidence intervals for the population mean.**

- z-score of 92% confidence interval is: 1.75
- z-score of 96% confidence interval is: 2.05
- z-score of 98% confidence interval is: 2.33

In [13]:
def population_mean(x, z):
    return [np.mean(x)-z*(np.std(x)/math.sqrt(len(x))), np.mean(x)+z*(np.std(x)/math.sqrt(len(x)))]

In [14]:
population_mean(population_45, 1.75)

[0.7342112456758711, 1.5057887543241286]

In [15]:
population_mean(population_45, 2.05)

[0.6680760306488778, 1.571923969351122]

In [16]:
population_mean(population_45, 2.33)

[0.6063498299570171, 1.6336501700429826]

**3. For each of the confidence intervals in step 2, use the information already calculated to determine the sample size needed to estimate the population mean to within .05 units.**

Formula for the Minimum Sample Size Needed for an Interval Estimate of the Population Mean
$$n = (\frac{Z_{\alpha/2}\dot\sigma}{E})^2$$
where E is the margin of error. If necessary, round the answer up to obtain a whole number. That is, if there is any fraction or decimal portion in the answer, use the next whole number for sample size n.

$\alpha = 0.92$, $Z = 1.75, \sigma = 1.48$

In [101]:
(1.75*1.48/0.05)**2

741.5818240000002

$\alpha = 0.96$, $Z = 2.05, \sigma = 1.48$

In [102]:
(2.05*1.48/0.05)**2

3682.062399999999

$\alpha = 0.98$, $Z = 2.33, \sigma = 1.48$

In [103]:
(2.33*1.48/0.05)**2

4756.585023999998

### Part 3

**1. Use the random sampling method to generate a random sample of size 25 from the given International Migration Rates in the dataset. Then calculate the sample mean, sample standard deviation, sample size, degrees of freedom and sampling error.**

In [17]:
population_25 = np.random.choice(population, size = 25, replace = True)

In [18]:
print("The sample mean is %.5f"%np.mean(population_25))

The sample mean is 1.16800


In [85]:
print("The sample standard deviation is %.5f"%np.std(population_25))

The sample standard deviation is 1.86970


In [20]:
print("the sample variance is %.5f"%np.var(population_25))

the sample variance is 3.49578


In [21]:
print("the sample size is %.5f"%population_45.size)

the sample size is 45.00000


In [22]:
z_score = np.zeros(len(population_25))
z_score = (population_25 - np.mean(population_25))/np.std(population_25)
sampling_error = (np.std(population))/(math.sqrt(population_25.size))
print("the sampling error is %.5f"%sampling_error)

the sampling error is 0.37950


**2. Use the sample to calculate margin of error, upper and lower confidence interval limit for 92%, 96% and 98% confidence intervals for the population mean. You should know whether to use a z or t value.**

- z-score of 92% confidence interval is: 1.75
- z-score of 96% confidence interval is: 2.05
- z-score of 98% confidence interval is: 2.33

In [23]:
def population_mean(p, z, n):
    return math.sqrt(p*(1-p)/n)
n = len(population_25)

In [24]:
print("Margin of error for the sample in 92%% confidence interval is: %.5f"%population_mean(0.92, 1.75, n))

Margin of error for the sample in 92% confidence interval is: 0.05426


In [25]:
print("Margin of error for the sample in 96%% confidence interval is: %.5f"%population_mean(0.96, 2.05, n))

Margin of error for the sample in 96% confidence interval is: 0.03919


In [26]:
print("Margin of error for the sample in 98%% confidence interval is: %.5f"%population_mean(0.98, 3.33, n))

Margin of error for the sample in 98% confidence interval is: 0.02800


### Part 4

**1. Use the random sampling method to generate a random sample of size 150 from the given International Migration Rates in the dataset. Calculate the sample proportion that are less than 1 for the random sample**

In [27]:
population_150 = np.random.choice(population, size = 150, replace = True)

In [28]:
print("Sample proportion that are less than 1 for the random sample is %.5f"%(len(population_150[population_150<1])/len(population_150)))

Sample proportion that are less than 1 for the random sample is 0.62667


**2. Calculate the sample proportion of counties whose international migration rate (per 1000) is greater than 2.**

In [29]:
print("Sample proportion that are greater than 2 for the random sample is %.5f"%(len(population_150[population_150>2])/len(population_150)))

Sample proportion that are greater than 2 for the random sample is 0.22000


**3. Calculate the sample proportion of success, the sample proportion of failure, sample size and sampling error.**

In [30]:
print("Sample proportion of success is %.5f"%(len(population_150[population_150>2])/len(population_150)))

Sample proportion of success is 0.22000


In [31]:
print("Sample proportion of failure is %.5f"%(len(population_150[population_150<2])/len(population_150)))

Sample proportion of failure is 0.78000


In [32]:
z_score = np.zeros(len(population_150))
z_score = (population_150 - np.mean(population_150))/np.std(population_150)
sampling_error = (np.std(population))/(math.sqrt(population_150.size))
print("the sampling error is %.5f"%sampling_error)

the sampling error is 0.15493


**4. Using the sample to create a 90%, a 95%, and 99% confidence intervals for the population proportion of counties whose international migration rate (per 1000) is greater than 2 in the designated cells. Calculate the margin of error, upper and lower CI Limit and CI width.**

- z-score of 90% confidence interval is: 1.65
- z-score of 96% confidence interval is: 1.67
- z-score of 99% confidence interval is: 2.58

In [40]:
n = len(population_150)

In [41]:
print("Margin of error for the sample in 90%% confidence interval is: %.5f"%population_mean(0.90, 1.65, n))

Margin of error for the sample in 90% confidence interval is: 0.02449


In [42]:
print("Margin of error for the sample in 96%% confidence interval is: %.5f"%population_mean(0.96, 1.67, n))

Margin of error for the sample in 96% confidence interval is: 0.01600


In [43]:
print("Margin of error for the sample in 99%% confidence interval is: %.5f"%population_mean(0.99, 2.58, n))

Margin of error for the sample in 99% confidence interval is: 0.00812


**5. Determine the minimum sample size needed to estimate the population proportion of interest to within 8%. Do this for each confidence interval above. Assume that there is no prior knowledge about the proportion of counties whose international migration rate (per 1000) is greater than 2.**

Formula for Minimum Sample Size Needed for Interval Estimate of a Population Proportion:
$$n = \hat{p}\hat{q}(\frac{Z_{\alpha/2}}{E})^2$$

Since $Z_{\alpha/2} = 1.96$, E = 0.08 and $\hat{p}=0.2$, then
$$n = \hat{p}\hat{q}(\frac{Z_{\alpha/2}}{E})^2 = (0.2)(0.8)(\frac{1.96}{0.02})^2$$

In [100]:
(0.2)*(0.8)*(1.96/0.02)**2

1536.6400000000003

**6. Calculate the population proportion of success and the population proportion of failure.**

In [45]:
print("Sample proportion of success is %.5f"%(len(population[population>2])/len(population)))

Sample proportion of success is 0.18182


In [46]:
print("Sample proportion of failure is %.5f"%(len(population[population<2])/len(population)))

Sample proportion of failure is 0.80584


### Part 5

**1. Use the random sampling method to generate a random sample of size 22 from the given population.**

In [47]:
population_22 = np.random.choice(population, size = 22, replace = True)

**2. Calculate the sample variance, sample standard deviation, and degrees of freedom.**

In [49]:
np.var(population_22)

1.4197727272727276

In [51]:
np.std(population_22)

1.1915421634473233

In [52]:
df = len(population_22) - 1
df

21

**3. Use the sample to create 90%, 95% and 98% confidence intervals for the population variance and standard deviation.**

- 90% confidence interval, we have $\alpha = 0.10$ which gives 5% of the area at each end of the chi-square distribution. We find values of $x_{0.95}^2 = 11.591$ and $x_{0.05}^2 = 32.671$.

- 95% confidence interval, we have $\alpha = 0.05$ which gives 2.5% of the area at each end of the chi-square distribution. We find values of $x_{0.975}^2 = 10.283$ and $x_{0.025}^2 = 35.479$.

- 98% confidence interval, we have $\alpha = 0.02$ which gives 1% of the area at each end of the chi-square distribution. We find values of $x_{0.99}^2 = 8.897$ and $x_{0.01}^2 = 38.932$.

In [69]:
def population_variance(n, s2, x2_1,x2_2):
    return [math.sqrt((n-1)*s2/x2_1), math.sqrt((n-1)*s2/x2_2)]

In [74]:
low_var, high_var = population_variance(df, np.var(population_22), 11.591, 32.671)
low_std, high_std = math.sqrt(low_var), math.sqrt(high_var)
print("the population variance of 90%% sample confidence interval is between %.5f to %.5f"%(low_var, high_var))
print("the population standard deviation of 90%% sample confidence interval is between %.5f to %.5f"%(low_std, high_std))

the population variance of 90% sample confidence interval is between 1.56518 to 0.93227
the population standard deviation of 90% sample confidence interval is between 1.25107 to 0.96554


In [75]:
low_var, high_var = population_variance(df, np.var(population_22), 10.283, 35.479)
low_std, high_std = math.sqrt(low_var), math.sqrt(high_var)
print("the population variance of 95%% sample confidence interval is between %.5f to %.5f"%(low_var, high_var))
print("the population standard deviation of 95%% sample confidence interval is between %.5f to %.5f"%(low_std, high_std))

the population variance of 95% sample confidence interval is between 1.66175 to 0.89462
the population standard deviation of 95% sample confidence interval is between 1.28909 to 0.94584


In [76]:
low_var, high_var = population_variance(df, np.var(population_22), 8.897, 38.932)
low_std, high_std = math.sqrt(low_var), math.sqrt(high_var)
print("the population variance of 98%% sample confidence interval is between %.5f to %.5f"%(low_var, high_var))
print("the population standard deviation of 98%% sample confidence interval is between %.5f to %.5f"%(low_std, high_std))

the population variance of 98% sample confidence interval is between 1.78650 to 0.85403
the population standard deviation of 98% sample confidence interval is between 1.33660 to 0.92414


### Part 6

**Use the sample in Part 1 to test the hypothesis that the average migration rate (per 1000) in the population is greater than 1. Use alpha = 0.05.**

**Step 1:** 
$H_0: \mu = 1.244$ $H_1: \mu > 1$

**Step 2:**
Find the critical value. Since $\alpha = 0.05$ and the test is a right-tailed test, the critical value is $z = +1.65$

**Step 3:**
Compute the test value: $z = \frac{\bar{X} - \mu}{\sigma / \sqrt{n}}$

In [82]:
(np.mean(population) - 1)/(np.std(population)/math.sqrt(len(population_45)))

0.8620903774822735

**Step 4:** Make the decision. Since the test value 0.86 is smaller than the critical value 1.65, the decision is not to reject the null hypothesis.

**Step 5:** Summarize the results There is enough evidence to support the claim that the hypothesis that the average migration rate (per 1000) in the population is greater than 1. Use alpha = 0.05.

### Part 7

**Use the sample in Part 3 to test the hypothesis that the population average migration rate (per 1000) is less than 1.5. Use alpha = 0.02**

**Step 1:** 
$H_0: \mu = 1.168$ $H_1: \mu < 1.5$

**Step 2:**
Find the critical value. Since $\alpha = 0.02$ and the test is a right-tailed test, the critical value is $z = +2.05$

**Step 3:**
Compute the test value:

In [83]:
(np.mean(population) - 1)/(np.std(population)/math.sqrt(len(population_45)))

1.1680000000000001

**Step 4:** Make the decision. Since the test value 1.168 is smaller than the critical value 2.05, the decision is reject the null hypothesis.

**Step 5:** Summarize the results There is enough evidence to support the claim that the hypothesis that the population average migration rate (per 1000) is less than 1.5. Use alpha = 0.02

### Part 8

**Use the sample in part 4 to test the hypothesis that less than 20% of the US counties have a migration rate that is higher than 2 (in 1000). Use alpha = 0.03.**

**Step 1:** State the hypotheses and identify the claim.
$H_0: p = 0.2$ $H_1: p < 0.2$

**Step 2:**
Compute the test value.
$\hat{p} = \frac{X}{n} = 0.22$
$p = 0.2$ and $q = 1-p = 0.8$

**Step 3:**
Compute the test value:

In [90]:
(0.22-0.2)/(math.sqrt(0.22*0.2/len(population_150)))

1.1677484162422838

**Step 4:** Do not reject the null hypothesis, since the test value does not fall in the critical region (< -1.88).

**Step 5:** There is not enough evidence to reject the claim that less than 20% of the US counties have a migration rate that is higher than 2 in alpha = 0.03.

### Part 9

**Use the sample in Part 5 to test the hypothesis that the standard deviation of the population emigration rates (in 1000) is different from 1.2. Use alpha = 0.05.**

**Step 1:** State the hypotheses and identify the claim.
$H_0: \theta^2 = 1.2$ $H_1: p \neq 1.2$

**Step 2:** Find the critical values. Since this test is a two-tailed test at a 0.05, the critical values for 0.025 and 0.975 must be found. The degrees of freedom are 21; hence, the critical values are 35.479 and 10.283, respectively. 

**Step 3:** Compute the test value:
$$X^2 =\frac{(n-1)*s^2}{\sigma^2} = \frac{(22-1)(1.42)}{1.2^2} $$
 


In [98]:
(22-1)*(1.42)/(1.2**2)

20.708333333333336

**Step 4:** Make the decision. Do not reject the null hypothesis, since the test value falls between the critical values (10.283 < 20.71 < 35.479) and in the noncritical region.